parsec/core/gui/devices_widget.py

```diff
@@ -88,7 +93,13 @@ def __init__(self, core, jobs_ctx, event_bus, *args, **kwargs):
         self.list_error.connect(self.on_list_error)
         self.line_edit_search.textChanged.connect(self.filter_timer.start)
         self.filter_timer.timeout.connect(self.on_filter_timer_timeout)
-        self.reset()
+        self.initialized = False
+
+    def show(self):
+        if not self.initialized:
```

Comment:
I'm guessing that you're checking for initialization to avoid resetting the widget, but does it make that much of a difference? I usually avoid adding logic for performance reason because it has bitten us in the ass in the past.

I would also add that the guy who wrote the original code is a moron and it seems like the device list is never reset. Ideally we would like to reset() on an event (I have no idea if a `device.new` event exists) but resetting when shown would still be better than nothing.

Reply:
The original idea there is to avoid getting errors when you just log  because the widget can not be filled (Like the exception reported in the issue #1191 ).In fact now the central widget handles this event errors so it probably does not matter to call `reset` at the end of `__init__` .

The event approach you suggest songs to be the right thing to do.  With my initialized flag, If there is no error at the first display, the list is not going to be refilled on the second one. An easy fix is to remove this initialized flag and always requests devices on display.

About events, is it the `central_widget` that is in charge to listen to `device.new` event (or `user.new` in the case of a user event and the `user_widget`) and then call `reset` on the `devices_widget`, or the `devices_widget` has to listen itself the events_bus ? (For example, i think to call reset from the central widget if the revoked user event happened to clear the local cache)